Lib/test/test_dataclasses.py

```diff
@@ -1442,16 +1442,27 @@ class B:
                     replace(obj, x=0)
 
     def test_is_dataclass_genericalias(self):
+        T = TypeVar("T")
         @dataclass
         class A(types.GenericAlias):
             origin: type
             args: type
+        @dataclass
+        class B(Generic[T]):
+            a: T
+        @dataclass
+        class C(Generic[T]):
+            a: T
+            __class_getitem__ = classmethod(types.GenericAlias)
```

Comment:
The test for this class only passes because the `typing` module is already imported. We need a test for the case when it is not imported (it means that we cannot inherit from `Generic` and should temporary remove `typing` from `sys.modules`).

Reply:
Fixed, added another test to cover this case